sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -224,9 +224,13 @@ public ColumnarBatch resultBatch() {
 
   /**
    * Can be called before any rows are returned to enable returning columnar batches directly.
+   *
+   * @param maxColumns The maximum number of fields supported by whole stage codegen.
    */
-  public void enableReturningBatches() {
-    returnColumnarBatch = true;
+  public boolean tryEnableReturningBatches(int maxColumns) {
```

Comment:
Why are we pushing a config value into the reader so that it can switch?  Can't we just put this logic in the query planner where the decision to call `enableReturningBatches` is.  I think its clearer if there is a single place where we decide to use column batches or not.